src/Dfc.CourseDirectory.Web/Helpers/Attributes/SelectedProviderNeededAttribute.cs

```diff
@@ -15,7 +15,7 @@ public override void OnActionExecuting(ActionExecutingContext context)
             int qsUKPRN;
             Int32.TryParse(context.HttpContext.Request.Query["UKPRN"].ToString(), out qsUKPRN);
             var session = context.HttpContext.Session.GetInt32("UKPRN");
-            if (qsUKPRN != null && session == null)
+            if (session == null)
```

Comment:
Seems like we've lost a check here... any particular reason? Should we not be checking that `qsUKPRN` has a value?